tests/engines/registries/unit/services/share-search-test.ts

```diff
@@ -80,4 +80,19 @@ module('Registries | Unit | Service | share-search', hooks => {
         assert.equal(registrations[0].contributors[0].name, 'Graham > Berlin');
         assert.equal(registrations[0].contributors[1].name, 'Nicole < Grant');
     });
+
+    test('recognizes all OSF source envs', function(this: TestContext, assert) {
+        const service = this.owner.lookup('service:share-search') as ShareSearch;
+        [
+            'https://osf.io/w4yhb/',
+            'https://test.osf.io/w4yhb/',
+            'https://staging.osf.io/w4yhb/',
+            'https://staging2.osf.io/w4yhb/',
+            'https://staging3.osf.io/w4yhb/',
+        ].forEach(mainLink => {
+            ES_RESPONSE.hits.hits[0]._source.identifiers = [mainLink];
```

Comment:
Minor: It's not great to modify `ES_RESPONSE` in place, since that could mess with other tests based on the order they run. Maybe replace it with a function `getSearchResponse(identifiers: string[])`?